app/interactions/matches/create_match.rb

```diff
@@ -0,0 +1,88 @@
+class CreateMatch < ActiveInteraction::Base
+  integer :number_of_players, default: 6
+
+  validates_numericality_of :number_of_players,
+                            greater_than_or_equal_to: 3,
+                            less_than_or_equal_to: 6
+
+  def execute
+    ActiveRecord::Base.transaction do
+      match = Match.create!
+      match.create_board!
+      create_players(match, number_of_players)
+      # create_power_pool(match)
+      # create_decks(match, use_tides_of_battle_cards)
+      match
+    end
+  end
+
+  private
+
+  # 3 players: Baratheon Lannister Stark
+  # 4 players: Baratheon Lannister Stark Greyjoy
+  # 5 players: Baratheon Lannister Stark Greyjoy Tyrell
+  # 6 players: Baratheon Lannister Stark Greyjoy Tyrell Martell
+  # (Rulebook, page 28)
+  def create_players(match, number_of_players)
+    house_names = Player::HOUSES.first(number_of_players)
+    house_names.each do |house_name|
+      player = match.players.create!(match: match, house: house_name)
+      distribute_player_things(player, match.board)
+    end
+  end
+
+  def distribute_player_things(player, board)
+    create_orders(player, board)
+    create_power_tokens(player)
+    compose(CreateUnits, player: player)
+    compose(CreateHouseCards, player: player)
+  end
+
+  def create_tracks(match, board, players)
+    [
+      WildlingsTrack,
+      VictoryTrack,
+      # SupplyTrack,               # TODO
+      # RoundTrack,                # TODO
+      # KingsCourtInfluenceTrack,  # TODO
+      # IronThroneInfluenceTrack,  # TODO
+      # FiefdomsInfluenceTrack     # TODO
+    ].each do |track|
+      track.setup!(match, board, players)
+    end
+  end
+
+  def create_decks(match, use_tides_of_battle_cards)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Unused method argument - `match`. If it's necessary, use `_` or `_match` as an argument name to indicate that it won't be used. You can also write as `create_decks(*)` if you want the method to accept any arguments but don't care about them.](https://www.codacy.com/app/Got-Board-Team/got-board-api/file/2384555189/issues/source?bid=3216569&fileBranchId=3216571#l55)